src/main/java/fi/okm/jod/yksilo/service/profiili/Mapper.java

```diff
@@ -50,37 +64,31 @@ static OsaaminenDto mapOsaaminen(Osaaminen o) {
     return o == null ? null : new OsaaminenDto(URI.create(o.getUri()), o.getNimi(), o.getKuvaus());
   }
 
-  static YksilonOsaaminenDto mapYksilonOsaaminen(YksilonOsaaminen y) {
-    return y == null
+  static YksilonOsaaminenDto mapYksilonOsaaminen(YksilonOsaaminen entity) {
+    return entity == null
         ? null
         : new YksilonOsaaminenDto(
-            y.getId(),
-            mapOsaaminen(y.getOsaaminen()),
-            y.getLahde(),
-            mapToimenkuva(y.getToimenkuva()),
-            mapKoulutus(y.getKoulutus()));
+            entity.getId(), mapOsaaminen(entity.getOsaaminen()), mapOsaamisenLahde(entity));
+  }
+
+  private static OsaamisenLahdeDto mapOsaamisenLahde(YksilonOsaaminen entity) {
+    return new OsaamisenLahdeDto(entity.getLahde().getTyyppi(), entity.getLahde().getId());
   }
 
   static <T, U> Function<T, U> cachingMapper(Function<T, U> mapper) {
     final IdentityHashMap<T, U> mappingCache = new IdentityHashMap<>();
     return (T object) -> object == null ? null : mappingCache.computeIfAbsent(object, mapper);
   }
 
-  static List<YksilonOsaaminenDto> mapYksilonOsaaminen(List<YksilonOsaaminen> items) {
+  static List<YksilonOsaaminenDto> mapYksilonOsaaminen(Collection<YksilonOsaaminen> items) {
```

Comment:
Should check for not null?

Reply:
Good catch. Since this is a mapper, it should tolerate a null instead of failing.